soc/xtensa/intel_adsp/tools/cavstool.py

```diff
@@ -694,24 +735,21 @@ def handle(self):
 
         if action == CMD_DOWNLOAD:
             self.request.sendall(cmd)
-            recv_fn = self.request.recv(BUF_SIZE)
-            log.info(f"{self.client_address[0]} wrote: {recv_fn}")
+            recv_file = self.receive_fw()
 
-            try:
-                tmp_file = recv_fn.decode("utf-8")
-            except UnicodeDecodeError:
-                tmp_file = "zephyr.ri.decode_error"
-                log.info(f'did not receive a correct filename')
+            if recv_file:
+                self.request.sendall("success".encode('utf-8'))
+            else:
+                self.request.sendall("failed".encode('utf-8'))
+                log.error("Receive firmware failed.")
+
+            log.info("Flash completed. Ready to run...")
```

Comment:
I'm kinda confused, only the download was completed at this point, no? No flashing, like, writing firmware to memory where CPU will start running it, right?